specification/search/resource-manager/Microsoft.Search/preview/2019-10-01-preview/search.json

```diff
@@ -0,0 +1,1465 @@
+          "enum": [
+            "free",
+            "basic",
+            "standard",
+            "standard2",
+            "standard3",
+            "storage_optimized_l1",
+            "storage_optimized_l2"
+          ],
+          "x-ms-enum": {
+            "name": "SkuName",
+            "modelAsString": false
+          }
+        }
+      },
+      "description": "Defines the SKU of an Azure Cognitive Search Service, which determines price tier and capacity limits.",
+      "externalDocs": {
+        "url": "https://azure.microsoft.com/documentation/articles/search-sku-tier/"
+      }
+    },
+    "PrivateEndpointConnection": {
+      "properties": {
+        "id": {
+          "type": "string",
+          "description": "The ID of the private endpoint connection. This can be used with the Azure Resource Manager to link resources together.",
+          "externalDocs": {
+            "url": "https://aka.ms/search-linked-resources"
+          }
+        },
+        "name": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The name of the private endpoint connection.",
+          "externalDocs": {
+            "url": "https://aka.ms/search-naming-rules"
+          }
+        },
+        "type": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The resource type."
+        },
+        "properties": {
+          "$ref": "#/definitions/PrivateEndpointConnectionProperties"
+        }
+      }
+    },
+    "NetworkRuleSet": {
+      "properties": {
+        "endpointAccess": {
+          "type": "string",
+          "default": "Public",
+          "description": "The level of access to the search service endpoint. Public, the search service endpoint is reachable from the internet. Private, the search service endpoint can only be accessed via private endpoints. Default is Public.",
+          "enum": [
+            "Public",
+            "Private"
+          ],
+          "x-ms-enum": {
+            "name": "EndpointAccess",
+            "modelAsString": false
+          }
+        },
+        "ipRules": {
+          "type": "array",
+          "description": "A list of rules that define inbound network access to the search service endpoint. By default, the search service endpoint is reachable from the internet.",
+          "items": {
+            "type": "object",
```

Comment:
just wonder the reason behind this structure? `items` only contains one field.

Reply:
refactored to model the item object in #/definitions/.